sound/soc/sof/intel/hda-dai.c

```
@@ -364,7 +370,6 @@ static int non_hda_dai_hw_params(struct snd_pcm_substream *substream,
 	}
 
 	/* get stream_id */
-	ops = hda_dai_get_ops(substream, cpu_dai);
```

why move it up ? It is better to set ops near the usage of ops

---

because you don't want the hw_params to allocate anything if the ops is borked. It's intentional, you want to fail early and fail big, and make the error handling as simple as possible.